pkg/netpol/eval/resources.go

```diff
@@ -203,6 +207,54 @@ func (pe *PolicyEngine) upsertNamespace(ns *corev1.Namespace) error {
 	return nil
 }
 
+// checkLegalOwnerRefNameToPodLabels returns error if there are pod resources with same ownerReferences name but different labels
+func (pe *PolicyEngine) checkLegalOwnerRefNameToPodLabels(newPod *k8s.Pod) error {
+	if newPod.Owner.Name == "" { // the new pod does not have owner references
+		return nil
+	}
+	if _, ok := pe.ownersToLabelsMap[newPod.Namespace]; !ok { // add the new namespace to the map
+		pe.ownersToLabelsMap[newPod.Namespace] = make(map[string]*k8s.Pod)
+	}
+	firstPod, ok := pe.ownersToLabelsMap[newPod.Namespace][newPod.Owner.Name]
+	if !ok { // add the new ownerReference with this new pod
+		pe.ownersToLabelsMap[newPod.Namespace][newPod.Owner.Name] = newPod
+		return nil
+	}
+	// compare the owner first pod's labels with new pod's Labels
+	return diffBetweenPodsLabels(firstPod, newPod)
+}
+
+// helper: given two pods of same owner, returns an error if there are diffs between the pods' labels maps
+func diffBetweenPodsLabels(firstPod, newPod *k8s.Pod) error {
```

Comment:
better split to a function that handles the error messages and a function that computes the labels gap details.

Reply:
- currently I raise the error with first diff found, 
is replacing these lines of computing the error msg and raising it (e.g. line 240-241) by calling a def that handle error msg writing (e.g return computeErrMsg(...) is enough , 
or   do you prefer to return as result all differences found and after that write the error message with any?

Reply:
it is sufficient to return the first difference found